manifest.yml

```diff
@@ -0,0 +1,14 @@
+---
+applications:
+- name: ((watchtower_app_name))
+  buildpacks:
+    - go_buildpack
+  env:
+    CF_USER: ((cf_user))
+    CF_PASS: ((cf_pass))
+    CLOUD_CONTROLLER_URL: "https://api.fr.cloud.gov"
+    REFRESH_INTERVAL: 5m
+  processes:
+  - type: web
+    instances: 1
+    memory: 64m
```

Comment:
Two comments on this file. I worry about over-using variables here, especially in an example. I haven't found it necessary to parameterize the application name or refresh interval during usage of Watchtower. Users can do this if they choose, but I think it's probably best to leave those as some example value rather than trying to demonstrate too many things with the examples here since the point is "you can use this as a starting place".

Other than that, I think this manifest file can be simplified. The `memory` and `instances` elements can be moved to the main app definition and the `processes` section can be eliminated as seen in the suggestion. The simplest use-case is running watchtower as a single-process app that scales horizontally for HA, and structuring the example manifest like this should help newer CloudFoundry users not have to dig into too much of the details of manifest file definitions.

Additionally, for production, I'm not sure whether the build-from-source method here is best, or whether we should point users to the binary_buildpack, and run a build from a pinned release of Watchtower. See Watchtower's [releases page](https://github.com/18F/watchtower/releases) where you'll find pre-built binaries for many systems that can be used, as well as hashes that can be validated by users who download the binary prior to running.
```suggestion
---
applications:
- name: my-watchtower-app
  instances: 2
  memory: 64m
  buildpacks:
    - go_buildpack
  env:
    CF_USER: ((cf_user))
    CF_PASS: ((cf_pass))
    CLOUD_CONTROLLER_URL: "https://api.fr.cloud.gov"
```

Reply:
I think I'd rather keep the app_name as a variable, as a flag that the user _must_ set a value. Using a hardcoded value will likely lead to people not changing it, and then getting a used route error. Other than that I'm fine with implementing these suggestions